src/main/java/beam/matsim/MatsimConfigUpdater.java

```diff
@@ -0,0 +1,36 @@
+package beam.matsim;
+
+import beam.sim.BeamConfigChangesObservable;
+import beam.sim.config.BeamConfig;
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import org.matsim.core.config.groups.ControlerConfigGroup;
+import org.matsim.core.config.groups.LinkStatsConfigGroup;
+import scala.Tuple2;
+
+import java.util.Observable;
+import java.util.Observer;
+
+@Singleton
+public class MatsimConfigUpdater implements Observer {
+    private ControlerConfigGroup controlerConfigGroup;
+    private LinkStatsConfigGroup linkStatsConfigGroup;
+
+    @Inject
+    MatsimConfigUpdater(BeamConfigChangesObservable beamConfigChangesObservable, LinkStatsConfigGroup linkStatsConfigGroup, ControlerConfigGroup controlerConfigGroup) {
+        this.controlerConfigGroup = controlerConfigGroup;
+        this.linkStatsConfigGroup = linkStatsConfigGroup;
+
+        beamConfigChangesObservable.addObserver(this);
+    }
+
+    @Override
+    public void update(Observable observable, Object o) {
+        Tuple2 t = (Tuple2) o;
```

Comment:
Is this guaranteed to be the objects you are casting? Should cast checks be made?

Reply:
looks like it is guaranteed, at least the same code appears in other places, but type checks when casting it is not a bad idea.
I will add it